unified-runtime/source/loader/ur_adapter_registry.hpp

```diff
@@ -294,7 +294,8 @@ class AdapterRegistry {
       for (const auto &device : manifest.device_types) {
         ur_device_tuple single_device = {manifest.backend, device};
 
-        auto matchesFilter = [single_device](const FilterTerm &f) -> bool {
+        const auto &matchesFilter =
```

Comment:
don't think we want this to be a reference, again not sure how this works semantically with lambdas but we're initializing a variable here, not storing a reference to something that already exists
```suggestion
        const auto matchesFilter =
```

Reply:
looks like it is valid to do this https://en.cppreference.com/w/cpp/language/reference_initialization (see "indirect binding") but I don't think there's any advantage to it since it falls back to a copy constructor, and I think it looks kind of weird

Reply:
sure, happy to update this